app/src/userpages/components/ProfilePage/ProfileSettings/index.jsx

```diff
@@ -62,57 +65,32 @@ export class ProfileSettings extends Component<Props> {
         return (
             <Fragment>
                 <Form onSubmit={this.onSubmit}>
-                    <FormGroup>
-                        <Label>
-                            Email
-                        </Label>
-                        <div>{this.props.user && this.props.user.username}</div>
-                    </FormGroup>
-                    <FormGroup>
-                        <Label>
-                            Password
-                        </Label>
-                        <div>
-                            <ChangePassword.Button />
-                        </div>
-                    </FormGroup>
-                    <FormGroup>
-                        <Label>
-                            Full Name
-                        </Label>
-                        <Input
+                    <div className={styles.fullname}>
+                        <TextInput
+                            label="Your name"
                             name="name"
                             value={this.props.user ? this.props.user.name : ''}
                             onChange={this.onNameChange}
                             required
+                            preserveLabelSpace
                         />
-                    </FormGroup>
-                    <FormGroup>
-                        <Label>
-                            Timezone
-                        </Label>
-                        <Select
-                            placeholder="Select timezone"
+                    </div>
+                    <div className={styles.email}>
+                        <TextInput label="Email" value={(this.props.user && this.props.user.username) || ''} readOnly />
```

Comment:
There's a couple of `this.props.user && this.props.user.<something>` around here. Maybe, since we're touching it, you could give `this.props.user` a const? Maybe even `const user = this.props.user || {}` so there's no need (I hope) to check if user is defined. :)